### ISSUE TITLE
It is recommended to replace third-party deform_conv2d with torchvision.ops.deform_conv2d.

### ISSUE THREAD

**Issue by PatrickZoo** [YufeiWang777/LRRU#6]
Due to various version reasons between OS, CUDA, g++, and PyTorch, it is often difficult to compile and install

third-party DeformConv source code. This can be frustrating, as it can lead to various compilation errors.

Since the publication of the NLSPN paper, the Depth Completion community has increasingly used DeformConv.

After our experiments, we found that the official DeformConv version of PyTorch is effective.

For the LRRU source code, two changes are required:

```python
### model/model_dcnv2.py ###

### Line 8 ###
# from model.dcn_v2 import dcn_v2_conv
from torchvision.ops import deform_conv2d


### Line 229 to Line 231 ###
# output = dcn_v2_conv.apply(
#     depth, offset, weight, self.w, self.b, self.stride, self.padding,
#     self.dilation, self.deformable_groups)
output = deform_conv2d(
      depth, offset, weight=self.w, bias=self.b, stride=self.stride,
      padding=self.padding, dilation=self.dilation, mask=weight)
```

After our tests, the pre-trained model LRRU_Base.pt achieved an RMSE of 729.5, which is consistent with the

results reported in the author's paper.

Thank you to the author for their excellent work.

**Comment by YufeiWang777** [comment]
Thanks for your very valuable advice! Since the version of torchvision in my environment is 0.8.2, I don't test the code using torchvision.ops.deform_conv2d. Do you obtain the RMSE (729.5) of LRRU_Base using the revised code?

**Comment by PatrickZoo** [comment]
Except for the above two code changes, the rest of the code remains consistent with the released code.

Our system running environment is as follows:
- Ubuntu: 16.04
- CUDA: 10.2
- Python: 3.8.17
- PyTorch: 1.12.1
- torchvision: 0.13.1
- cuDNN: 7.6.5

The test results using the `configs/val_lrru_base_kitti.yml` are as follows:

- Without TTA(Test Time Augmentation):

  ```
  test_model: ./pretrained/LRRU_Base.pt 
  test_option: val 
  val:val 
  test_name: ben_depth 
  test_not_random_crop: False 
  tta: False
  
  0000 |  test_Metric|  RMSE: 0.7295  MAE: 0.1888  iRMSE: 0.0019  iMAE: 0.0008  REL: 0.0101  D^1: 0.9982  D^2: 0.9995  D^3: 0.9998  
  Elapsed time : 144.09094905853271 sec, Average processing time : 0.1440909490585327 sec
  ```

- With TTA(Test Time Augmentation):

  ```
  test_model: ./pretrained/LRRU_Base.pt 
  test_option: val 
  val:val 
  test_name: ben_depth 
  test_not_random_crop: False 
  tta: True
  
  0000 |  test_Metric|  RMSE: 0.7234  MAE: 0.1881  iRMSE: 0.0019  iMAE: 0.0008  REL: 0.0100  D^1: 0.9983  D^2: 0.9995  D^3: 0.9998  
  Elapsed time : 287.8145065307617 sec, Average processing time : 0.28781450653076174 sec
  ```

It is important to note that DeformConv V2 adds a modulated factor compared to the original DeformConv. 
In some third-party implementations, this factor is called "weight", which can be confusing as it is also the name
of the convolution weights. In the torchvision implementation, this factor is called "mask", so it is important to be
aware of this difference when passing parameters.

**Comment by YufeiWang777** [comment]
Cool! I will modify the readme based on your suggestions. Thanks for your contribution.
